java/CMakeLists.txt

```diff
@@ -45,13 +48,13 @@ target_include_directories(vw_jni PUBLIC
   ${JNI_INCLUDE_DIRS})
 
 # Ensure target directory exists
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/target/)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/target/bin/natives/linux_64/)
 add_custom_command(TARGET vw_jni POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:vw_jni> ${CMAKE_CURRENT_SOURCE_DIR}/target/bin/
+    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:vw_jni> ${CMAKE_CURRENT_SOURCE_DIR}/target/bin/natives/linux_64/
 )
 
 # enable-new-dtags and rpath enables shared object library lookup in the location of libvw_spark_jni.so
-target_link_libraries(vw_jni PUBLIC -Wl,--enable-new-dtags -Wl,-rpath,\"\$ORIGIN\" vw)
+target_link_libraries(vw_jni PUBLIC -static-libgcc -static-libstdc++ -fPIC -Wl,--enable-new-dtags -Wl,-rpath,\"\$ORIGIN\" vw)
```

Comment:
Do we _always_ want to be linking static and with fPIC? I can understand fPIC but it seems we want to check for STATIC_LINK_VW_JAVA for the former

Reply:
added the check. thanks for the feedback.